src/components/DateRangeInput/OverlayComponent.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import dayjs from 'dayjs';
+import styled from '@emotion/styled';
+import Box from '../Box';
+import Card from '../Card';
+import Button from '../Button';
+import Flex from '../Flex';
+import TimePicker from './TimePicker';
+import { IconButton } from '../../index';
+
+interface CustomOverlayProps {
+  month?: Date;
+  date?: { from: Date; to: Date };
+  selectedDay?: Date;
+  withTime?: boolean;
+  goToNextMonth: () => void;
+  onCancel: () => void;
+  onApply: (date?: Date) => void;
+  onTimeUpdate: (date: { from?: Date; to?: Date }) => void;
+}
+
+const Wrapper = styled(Box)`
+  position: relative;
+
+  .DayPicker-Day {
+    position: relative;
+    z-index: 3;
+    .day {
+      position: relative;
+      z-index: 4;
+    }
+    .bg,
+    .highlight {
+      position: absolute;
+      top: 0;
+      left: 0;
+      height: 100%;
+    }
+    .bg {
+      width: 100%;
+      z-index: 2;
+    }
+    .highlight {
+      z-index: 1;
+      width: 50%;
+    }
+  }
+
+  .DayPicker-Day--start,
+  .DayPicker-Day--end {
+    background-color: transparent !important;
+    border-radius: 0 !important;
+    .bg {
+      border-radius: 50%;
+      background-color: ${({ theme }) => theme.colors['blue-400']};
+    }
+  }
+
+  .DayPicker-Day--start {
+    .highlight {
+      background-color: #2d3e54;
+      left: 50%;
+    }
+  }
+
+  .DayPicker-Day--end {
+    .highlight {
+      background-color: #2d3e54;
+      left: 0;
+    }
+  }
+
+  .DayPicker-Day--outside + .DayPicker-Day--end,
+  .DayPicker-Day--start.DayPicker-Day--end {
+    .highlight {
+      display: none;
+    }
+  }
+  .DayPicker-Week {
+    .DayPicker-Day--selected:not(.DayPicker-Day--start):not(.DayPicker-Day--end):not(.DayPicker-Day--outside) {
+      &:first-child {
+        border-radius: 4px 0 0 4px;
```

Comment:
We need to add border-radius for week views and since each row/week is a faux table-row `border-radius` and hidden overflows are not properly working.